src/app/games/tafl/TaflRules.ts

```diff
@@ -0,0 +1,462 @@
+            return (c.y === 0) || (c.y === fin);
+        } else if (c.x === fin) {
+            return (c.y === 0) || (c.y === fin);
+        }
+        return false;
+    }
+    public tryCapture(player: Player, landingPawn: Coord, d: Orthogonal, state: S): Coord {
+        /* landingPawn is the piece that just moved
+         * d the direction in witch we look for capture
+         * return the captured coord, or null if no capture possible
+         * 1. the threatened case dont exist         -> no capture
+         * 2: the threatened case is not an opponent -> no capture
+         * 3: the threatened case is a king -> delegate calculation
+         * 4: the threatened case is a pawn -> delegate calculation
+         */
+        const threatened: Coord = landingPawn.getNext(d);
+        if (!threatened.isInRange(this.config.WIDTH, this.config.WIDTH)) {
+            return null; // 1: the threatened case dont exist, no capture
+        }
+        const threatenedPawnOwner: RelativePlayer = this.getRelativeOwner(player, threatened, state.board);
+        if (threatenedPawnOwner !== RelativePlayer.OPPONENT) {
+            return null; // 2: the threatened case is not an opponent
+        }
+        if (state.board[threatened.y][threatened.x].isKing()) {
+            return this.captureKing(player, landingPawn, d, state);
+        }
+        return this.capturePawn(player, landingPawn, d, state);
+    }
+    private captureKing(player: Player, landingPiece: Coord, d: Orthogonal, state: S) : Coord {
+        /**
+         * the king is the next coord after c (in direction d)
+         * the landingPiece partipate in the capture
+         *
+         *  1: allied is out-of-range
+         *      2: if two other are invaders AND LEGAL                  -> capture king (1 border + 3 invaders)
+         *      3: if one invaders and one empty throne
+         *          3.1: if king capturable by empty-throne and borders -> capture king (1 border, 1 throne, 2 invaders)
+         *  4: back is empty
+         *      5: if back is not a throne                              -> no capture
+         *      here, back is an empty throne
+         *      6: if king not capturable by empty throne               -> no capture
+         *      7: if king capturable by 2                              -> capture king (1 invader + throne)
+         *      8: else if two-other-coord are invader                  -> capture king (3 invaders + throne)
+         *  9: allied is an invader
+         *     10: if king is capturable by two                         -> capture king (2 invaders)
+         *     11: if 2 others around king are invaders                 -> capture king (4 invaders)
+         * So these are the different victory way for the invaders :
+         * - 2 invaders
+         * - 1 invaders 1 empty-throne
+         * - 3 invaders 1 throne
+         * - 2 invaders 1 throne 1 border
+         * - 3 invaders 1 border
+         * - 4 invaders
+         */
```

Comment:
dispatch comment in sub-functions ?

Reply:
meh; It's better without it no ? The code is far more clear now, thank to the tetragrammaton !

Reply:
yep, remove comment then!